protoc-gen-grpc-gateway/descriptor/types.go

```diff
@@ -249,15 +249,37 @@ func (p FieldPath) AssignableExpr(msgExpr string) string {
 	if l == 0 {
 		return msgExpr
 	}
-	components := []string{msgExpr}
+
+	prepare := []string{}
+	fqnm := msgExpr
 	for i, c := range p {
+		// Check if it is a oneOf field. if
+		// This field is one from Oneof definition, we can not assign value directly.
+		if c.Target.FieldDescriptorProto.OneofIndex != nil {
```

Comment:
`c.Target.OneofIndex` is simpler.

Similarly, `msg := c.Target.Message`

Reply:
thanks.